### PR TITLE
Refactor authorized request checks

### PR DESCRIPTION
**Pull request by 2HEADSs**
Validate the respective user against the request through external checks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -157,35 +157,35 @@ Update entry with ID `654651caf696083cab72ab1c` in the `events` collection:
...
+    Changes: Only the administrator who is approved and not deleted can change the role of an individual user.
     To fulfill the request, the admin must send role. If new role is 'organizer', the fields 'organizatorName' and 'phone' also should be fulfilled if there was empty before update.
 
 -   **Method:** `PUT`
--   **Endpont:** `/user/approveOrganizer/:id`
+-   **Endpont:** `/user/approveUser/:id`
     Headers: `Content-Type: application/json`
     Body: JSON-formatted data
-    Changes: Admin which is approved and not deleted, only can approve/disapprove the organizer which also need to be not deleted.
+    Changes: Only the administrator who is approved and not deleted can approve/dissapprove an individual user.
```
Comment (Getsov) [inline README.md:181]: "disapprove"

File: src/controllers/userController.js [hunk 2]
```diff
@@ -181,10 +169,14 @@ userController.get('/getMyEvents', async (req, res) => {
     }
 });
 
-userController.get('/getMyFavourites', async (req, res) => {
+// Approving / Unapproving user/organizer
```
Comment (Getsov) [inline src/controllers/userController.js:172]: here is "unapproving" , above was "disapproving", which one will last?

File: src/services/userService.js [hunk 3]
```diff
@@ -165,51 +134,50 @@ async function editUserRole(idOfUserForEdit, requestBody, requesterId) {
...
-    if (!userForEdit.isApproved || !requester.isApproved) {
-        throw new Error('This profile is not approved!');
-    }
 
     if (requestBody.role == 'organizer') {
-        if (userForEdit.organizatorName == '' && userForEdit.phone == '') {
-            if (!requestBody.organizatorName || !requestBody.phone) {
+        if (userForEdit.organizatorName == '') {
+            if (!requestBody.organizatorName) {
                 throw new Error('Fill all required fields!');
```
Comment (Getsov) [inline src/services/userService.js:149]: More specified errors -> "Organizator's name is missing" for example

File: src/services/userService.js [hunk 4]
```diff
@@ -165,51 +134,50 @@ async function editUserRole(idOfUserForEdit, requestBody, requesterId) {
...
+            } else {
+                userForEdit.organizatorName = requestBody.organizatorName;
+            }
+        }
+
+        if (userForEdit.phone == '') {
+            if (!requestBody.phone) {
+                throw new Error('Fill all required fields!');
```
Comment (Getsov) [inline src/services/userService.js:157]: More specified errors -> "Phone is missing" for example

File: src/services/userService.js [hunk 5]
```diff
@@ -385,6 +314,40 @@ async function getAllUsers(requesterId) {
     const allUsers = await User.find();
     return allUsers;
 }
+async function addEventToLikedEvents(eventId, userId, isAlreadyLiked) {
```
Comment (Getsov) [inline src/services/userService.js:317]: addRemoveLikedEvent

File: src/utils/securityCheck.js [hunk 6]
```diff
@@ -1,22 +1,36 @@
...
+    }
 
-}
+    //If the requester is not "admin" or is not the user itslef
+    if (!isAdmin && requester._id.toString() != userForEdit._id.toString()) {
+        throw new Error('You do not have rights to modify the record!');
+    }
+
+    //If the user for edit have property "isDeleted" : true
+    if (userForEdit.isDeleted) {
```
Comment (Getsov) [inline src/utils/securityCheck.js:27]: What if an admin wants to edit isDeleted property of a deleted user?

File: src/utils/securityCheck.js [hunk 7]
```diff
@@ -1,22 +1,36 @@
...
+        throw new Error('This profile is deleted!');
+    }
+
+    //If the user for edit have property "isApproved" : false
+    if (!userForEdit.isApproved) {
```
Comment (Getsov) [inline src/utils/securityCheck.js:32]: What if an admin wants to edit isApproved property of an unapproved user?

File: src/services/userService.js (hunk 5 above)
Comment (2HEADSs) [reply src/services/userService.js:317]: fixed

File: src/utils/securityCheck.js (hunk 6 above)
Comment (2HEADSs) [reply src/utils/securityCheck.js:27]: Admin requests do not go into this file. Their checks are carried out at the service level.

File: src/utils/securityCheck.js (hunk 7 above)
Comment (2HEADSs) [reply src/utils/securityCheck.js:32]: Admin requests do not go into this file. Their checks are carried out at the service level.

File: src/services/userService.js (hunk 4 above)
Comment (2HEADSs) [reply src/services/userService.js:157]: fixed

File: src/services/userService.js (hunk 3 above)
Comment (2HEADSs) [reply src/services/userService.js:149]: added

File: src/controllers/userController.js (hunk 2 above)
Comment (2HEADSs) [reply src/controllers/userController.js:172]: fixed

File: README.md (hunk 1 above)
Comment (2HEADSs) [reply README.md:181]: Will remain as disapprove.

**Review by Getsov — APPROVED** [review]

**Pull request merged**